### PR TITLE
Refactor node constructors and misc code cleanup

### PR DESCRIPTION
**Pull request by t1m0thyj**
## Proposed changes

⚠️ This refactoring was done to facilitate support for custom encodings. I split the changes out into a separate PR to make them easier to review. This PR should be merged before #2594.

* Moved `openPS` method from Data Set Actions to `ZoweDatasetNode.openDs` to be consistent with USS
* Refactored Zowe tree node constructors to use interfaces which is cleaner than passing lots of arguments
  * Changed constructors in Zowe Explorer package, but not in ZE API, to avoid breaking changes
* Removed `@ts-nocheck` comment from `ZosJobNode` and fixed unchecked type errors
  * Removed `tooltip` getter/setter from `ZosJobNode` (since overriding property with getter/setter is invalid TS)
* Removed `profileName` from `ZoweUssNode` constructor since it's redundant with the `profile` object
* Removed duplicate unit tests for `UssTree`
* Removed unused source code in `packages/zowe-explorer/src/generators/messages`

## Release Notes

Milestone: 2.14.0

Changelog: N/A since these changes are not user-facing

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [ ] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [x] There is coverage for the code that I have added
- [ ] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

### CODE REVIEW

**Review by traeok — APPROVED** [review]
Tested and LGTM! Thanks @t1m0thyj 

I like these new constructors, definitely more human-readable 😋

**Review by awharn — APPROVED** [review]

**Review by zFernand0 — APPROVED** [review]
LGTM! 😋

File: packages/zowe-explorer/package.json [hunk 1]
```diff
@@ -935,16 +935,6 @@
...
-        {
-          "when": "view == zowe.uss.explorer && viewItem =~ /^textFile.*/",
-          "command": "zowe.uss.binary",
-          "group": "002_zowe_ussSystemSpecific@1"
-        },
-        {
-          "when": "view == zowe.uss.explorer && viewItem =~ /^binaryFile.*/",
-          "command": "zowe.uss.text",
-          "group": "002_zowe_ussSystemSpecific@2"
-        },
```
Comment (JillieBeanSim) [inline packages/zowe-explorer/package.json:947]: @t1m0thyj I see these right-click actions being removed from the USS menu.  Is there a new way to achieve this functionality in place that I am missing?  I have used this functionality a lot over the years and would really miss having it.

Comment (JillieBeanSim) [reply packages/zowe-explorer/package.json:947]: nm, see PR that introduces the new functionality

**Review by JillieBeanSim — APPROVED** [review]

**Pull request merged**